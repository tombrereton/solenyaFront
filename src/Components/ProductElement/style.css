.ProductElement {
  /* display: flex; */
  position: relative;
  margin: 1.5em;
}

.productContainer{
  display: flex;
  flex-wrap: wrap;
  margin-top: 60px;
  margin-bottom: 60px;
  justify-content: center;
}

.productDiscount,
.productPrice,
.productPriceSlashed {
  font-weight: 700;
}

.productPriceSlashed {
  color: #d0021b;
  text-decoration: line-through;
}

.productLink {
  font-weight: 500;
  text-decoration: none;
}

.productLabel > a {
  color: black;
}

.divLink {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-left: 15%;
  padding-right: 5%;
  font-size: 15px;
  font-family: "futura-pt";
  text-decoration: none;
  font-weight: 500;
  color: black;
}

.productLabelContainer {
  display: flex;
  position: absolute;
  justify-content: center;
  top: 10%;
  background: rgba(255, 255, 255, 0.5);
}

.imageCard {
  position: relative;
  display: flex;    
}

@media screen and (min-width: 1122px) {
  .ProductElement {
    width: auto;
  }

  img {
    width: 60%;
  }
  
  .productLabelContainer{
    left: 55%;
    height: 50%;
    margin-top: 10%;
  }

}

@media screen and (min-width: 1635px) {

  .ProductElement:nth-child(6n+4) > .imageCard > a,
  .ProductElement:nth-child(6n+5) > .imageCard > a,
  .ProductElement:nth-child(6n+6) > .imageCard > a{
    display: flex;
    flex-direction: row-reverse;
    height: 400px;
    }

  .ProductElement:nth-child(6n+4)  >  .productLabelContainer,
  .ProductElement:nth-child(6n+5)  >  .productLabelContainer,
  .ProductElement:nth-child(6n+6)  >  .productLabelContainer
   {
      left: 0%;
      right: 45%;
      width: 230px;
  } 
}


@media screen and (min-width: 1122px) and (max-width: 1635px){

  .ProductElement:nth-child(4n+3) > .imageCard > a,
  .ProductElement:nth-child(4n+4) > .imageCard > a  {
    display: flex;
    flex-direction: row-reverse;
    height: 400px;
    }

  .ProductElement:nth-child(4n+3)  >  .productLabelContainer,
  .ProductElement:nth-child(4n+4)  >  .productLabelContainer
   {
      left: 0%;
      right: 45%;
      width: 230px;
  } 


}


@media screen and (max-width: 1121px){
  .ProductElement {
    width: auto;
  }

  img {
    width: 100%;
  }

  .imageCard {  
    width: 53%; 
  }

  .ProductElement:nth-child(even) {
    display: flex;
    flex-direction: row-reverse;
    }

  .ProductElement:nth-child(even) > .productLabelContainer {
      left: 0%;
      right: 45%;
  } 

  .productContainer{
    margin-bottom: 35px;
  }

  .productLabelContainer {
    left: 45%;
    height: 80%;
 }

}
