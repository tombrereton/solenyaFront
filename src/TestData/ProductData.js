let ProductData = [
  {
    ProductId: 1,
    ProductName: "Warehouse Side Split Roll Neck Jumper",
    SplashImgUrl: "./product1/prod1-img1.jpg",
    Price: 4600,
    DiscountPrice: 0

  },
  {
    ProductId: 2,
    ProductName: "French Connection Checked Lined Harrington Jacket with Borg Collar",
    SplashImgUrl: "./product2/prod2-img1.jpg",
    Price: 6000,
    DiscountPrice: 0
  },
  {
    ProductId: 3,
    ProductName: "Adidas Originals Trefoil Hoodie In Grey",
    SplashImgUrl: "./product3/prod3-img1.jpg",
    Price: 5000,
    DiscountPrice: 0
  },
  {
    ProductId: 4,
    ProductName: "ASOS Body With Plunge Neck Long Sleeve And Thong",
    SplashImgUrl: "./product4/prod4-img1.jpg",
    Price: 1800,
    DiscountPrice: 1050

  },
  {
    ProductId: 5,
    ProductName: "All Saints Oversized Zip Biker Jacket in Leather",
    SplashImgUrl: "./product5/prod5-img1.jpg",
    Price: 42000,
    DiscountPrice: 0

  }
]

export default ProductData;
