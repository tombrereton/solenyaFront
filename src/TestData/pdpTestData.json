[
    {
    "ProductId": 3,
    "ProductName": "Adidas Originals Trefoil Hoodie",
    "SplashImgUrl": "https://i.imgur.com/vxfTdPz.jpg",
    "Price": 5000,
    "DiscountPrice": null,
    "Variants": [
    {
    "VariantId": 13
    },
    {
    "VariantId": 14
    },
    {
    "VariantId": 15
    },
    {
    "VariantId": 16
    },
    {
    "VariantId": 17
    },
    {
    "VariantId": 18
    },
    {
    "VariantId": 19
    },
    {
    "VariantId": 20
    },
    {
    "VariantId": 21
    },
    {
    "VariantId": 22
    },
    {
    "VariantId": 23
    },
    {
    "VariantId": 24
    }
    ],
    "ImageOptions": [
    {
    "Colour": "Grey",
    "ImageList": [
    "https://i.imgur.com/vxfTdPz.jpg",
    "https://i.imgur.com/Pm66fCS.jpg",
    "https://i.imgur.com/0wjsATF.jpg",
    "https://i.imgur.com/tI2n4lF.jpg"
    ]
    },
    {
    "Colour": "Black",
    "ImageList": [
    "https://i.imgur.com/fOslYDO.jpg",
    "https://i.imgur.com/ChyfUeg.jpg",
    "https://i.imgur.com/rAhgewy.jpg",
    "https://i.imgur.com/L6RuD5k.jpg"
    ]
    }
    ],
    "ProductDescription": "Hoodie by adidas Originals, Supplier code: BS2229, Soft-touch sweat, Drawstring hood, Raglan sleeves, Over-the-head style, Pouch pockets, Ribbed trims, Regular fit - true to size, Machine wash, Our model wears a size Medium and is 6'1‚Äù/185.5 cm tall",
    "ProductBrand": "Addidas Originals",
    "BrandDescription": "With a brand history stretching back over 60 years, adidas Originals draw inspiration from street culture and retro styles to provide fresh vintage inspired menswear. The adidas Originals range incorporates everything from the brands most iconic trainers to new vintage inspired clothes.",
    "Materials": "BFabric 1: 70% Cotton, 30% Polyester, Fabric 2: 97% Cotton, 3% Elastane, Lining: 100% Cotton.",
    "Gender": "Men"
    }
]