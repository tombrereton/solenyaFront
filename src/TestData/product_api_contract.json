"Product API contract, listings page: "
[ 
  {
    "productID": 1234,
    "productName":"Product1",
    "splashImgUrl":"url",
    "price":"£20.00",
    "discountPrice": "",
  }
]

"Product API contract, description page: "
[
   {
    "productID": 1234,
    "variants":
      [
        {
          "variantID": 123, 
          "images":
            [
              "image0":"image0",
              "image1":"image1",
              "image2":"image2",
              "image3":"image3"
            ],
        },
        {
          "variantID": 456, 
          "images":
            [
              "image0":"image0",
              "image1":"image1",
              "image2":"image2",
              "image3":"image3"
            ],
        },
      ]
    "productName":"Product1",
    "price":"£20.00",
    "discountPrice": "",
    "productDescription":"some text here",
    "productBrand":"some text here",
    "materials":"some text here",
    "brandDescription":"some text here",
  }
]

"Stock API contract: "
[
  {"variantID": 123, "size": "s", "colour" : "red", "quantity": 1},
  {"variantID": 456, "size": "m", "colour" : "black", "quantity": 3},
  {"variantID": 789, "size": "l", "colour" : "black", "quantity": 3},
]