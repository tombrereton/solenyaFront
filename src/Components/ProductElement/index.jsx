import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Header from "../Header/index.jsx";
import "./style.css";

const ProductElement = props => {
  return (
    <div className="ProductElement" style={{ margin: "1em" }}>
      <div className="imageCard">
        <Link
          id={"productImageId-" + props.ProductId}
          className="productLink"
          to={"products/" + props.ProductId}
          onClick={() =>
            Header.onPdpLinkClicked(
              props.ProductName,
              props.ProductId,
              "-WithImage"
            )}
        >
          <img width="150" src={props.SplashImgUrl} alt={props.ProductName} />
        </Link>
      </div>

      <div className="productLabelContainer">
        <div className="productLabel">
          <Link
            id={"productNameId-" + props.ProductId}
            className="productLink"
            to={"products/" + props.ProductId}
            onClick={() =>
              Header.onPdpLinkClicked(
                props.ProductName,
                props.ProductId,
                "-WithName"
              )}
          >
            {props.ProductName}
          </Link>
          {props.DiscountPrice === null ? (
            <div className="productPrice">
              £{(props.Price / 100).toFixed(2)}
            </div>
          ) : (
            <div>
              <div className="productPriceSlashed">
                £{(props.Price / 100).toFixed(2)}
              </div>
              <div className="productDiscount">
                £{(props.DiscountPrice / 100).toFixed(2)}
              </div>
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

ProductElement.propTypes = {
  ProductId: PropTypes.number.isRequired,
  SplashImgUrl: PropTypes.string.isRequired,
  Price: PropTypes.number.isRequired,
  ProductName: PropTypes.string.isRequired,
  DiscountPrice: PropTypes.number
};

export default ProductElement;
