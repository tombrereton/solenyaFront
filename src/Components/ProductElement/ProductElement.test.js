import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";
import ProductElement from "./";
import TestData from "../../TestData/ProductApiDataTest.json";

Enzyme.configure({ adapter: new Adapter() });

let dataElem = TestData[0];

console.error = message => {
  throw new Error(message);
};

describe("ProductElement", () => {
  it("It should match snapshot", () => {
    const wrapper = shallow(
      <ProductElement
        ProductId={dataElem.ProductId}
        ProductName={dataElem.ProductName}
        Price={dataElem.Price}
        SplashImgUrl={dataElem.SplashImgUrl}
        DiscountPrice={dataElem.DiscountPrice}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("the price of the product is not empty", () => {
    const card = (
      <ProductElement
        ProductId={dataElem.ProductId}
        ProductName={dataElem.ProductName}
        Price={dataElem.Price}
        SplashImgUrl={dataElem.SplashImgUrl}
        DiscountPrice={dataElem.DiscountPrice}
      />
    );
    const wrapper = shallow(card);
    const price = wrapper.find(".productPrice");
    const priceString = "£" + (dataElem.Price.toString() / 100).toFixed(2);
    expect(price.text()).toEqual(priceString);
  });

  it("the img url of the product is not empty", () => {
    const card = (
      <ProductElement
        ProductId={dataElem.ProductId}
        ProductName={dataElem.ProductName}
        Price={dataElem.Price}
        SplashImgUrl="foo"
        DiscountPrice={dataElem.DiscountPrice}
      />
    );
    const wrapper = shallow(card);
    const img = wrapper.find("img");
    expect(img.prop("src")).toEqual("foo");
  });

  it("requires a name prop", () => {
    expect(() =>
      shallow(
        <ProductElement
          Price={dataElem.Price}
          SplashImgUrl={dataElem.SplashImgUrl}
        />
      )
    ).toThrow();
  });

  it("requires a price prop", () => {
    expect(() =>
      shallow(
        <ProductElement
          ProductName={dataElem.ProductName}
          SplashImgUrl={dataElem.SplashImgUrl}
        />
      )
    ).toThrow();
  });

  it("requires a splash url prop", () => {
    expect(() =>
      shallow(
        <ProductElement
          ProductName={dataElem.ProductName}
          Price={dataElem.Price}
        />
      )
    ).toThrow();
  });

  it("the alt tag is not empty", () => {
    const card = (
      <ProductElement
        ProductName={dataElem.ProductName}
        Price={dataElem.Price}
        SplashImgUrl={dataElem.SplashImgUrl}
        DiscountPrice={dataElem.DiscountPrice}
      />
    );
    const wrapper = shallow(card);
    const img = wrapper.find("img");
    expect(img.prop("alt")).toEqual(dataElem.ProductName);
  });

  it("requires discount price displayed when not discount price null", () => {
    let discountPriceValue = 1200;
    const card = (
      <ProductElement
        ProductName={dataElem.ProductName}
        Price={dataElem.Price}
        SplashImgUrl={dataElem.SplashImgUrl}
        DiscountPrice={discountPriceValue}
      />
    );
    const wrapper = shallow(card);
    const discountPrice = wrapper.find(".productDiscount");
    expect(discountPrice.text()).toEqual("£12.00");
  });


  it("Should log the correct name click", () => {
    const wrapper = shallow(
      <ProductElement
        ProductId={dataElem.ProductId}
        ProductName={dataElem.ProductName}
        Price={dataElem.Price}
        SplashImgUrl={dataElem.SplashImgUrl}
        DiscountPrice={dataElem.DiscountPrice}
      />
    );
  });
});
