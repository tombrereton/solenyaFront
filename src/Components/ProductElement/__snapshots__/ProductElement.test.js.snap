// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductElement It should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductElement
    DiscountPrice={null}
    Price={4600}
    ProductId={123}
    ProductName="Warehouse Side Split Roll Neck Jumper"
    SplashImgUrl="./static/media/prod1-img1.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="imageCard"
        >
          <Link
            id="productImageId-123"
            onClick={[Function]}
            replace={false}
            to="products/123"
          >
            <img
              alt="Warehouse Side Split Roll Neck Jumper"
              src="./static/media/prod1-img1.jpg"
            />
          </Link>
        </div>,
        <Link
          className="divLink"
          id="productNameId-123"
          onClick={[Function]}
          replace={false}
          to="products/123"
        >
          <div
            className="productLabelContainer"
          >
            <div
              className="centeringContainer"
            >
              <div
                className="productLabel"
              >
                Warehouse Side Split Roll Neck Jumper
              </div>
              <div
                className="productPrice"
              >
                £
                46.00
              </div>
            </div>
          </div>
        </Link>,
      ],
      "className": "ProductElement",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            id="productImageId-123"
            onClick={[Function]}
            replace={false}
            to="products/123"
          >
            <img
              alt="Warehouse Side Split Roll Neck Jumper"
              src="./static/media/prod1-img1.jpg"
            />
          </Link>,
          "className": "imageCard",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="Warehouse Side Split Roll Neck Jumper"
              src="./static/media/prod1-img1.jpg"
            />,
            "id": "productImageId-123",
            "onClick": [Function],
            "replace": false,
            "to": "products/123",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Warehouse Side Split Roll Neck Jumper",
              "src": "./static/media/prod1-img1.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="productLabelContainer"
          >
            <div
              className="centeringContainer"
            >
              <div
                className="productLabel"
              >
                Warehouse Side Split Roll Neck Jumper
              </div>
              <div
                className="productPrice"
              >
                £
                46.00
              </div>
            </div>
          </div>,
          "className": "divLink",
          "id": "productNameId-123",
          "onClick": [Function],
          "replace": false,
          "to": "products/123",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="centeringContainer"
            >
              <div
                className="productLabel"
              >
                Warehouse Side Split Roll Neck Jumper
              </div>
              <div
                className="productPrice"
              >
                £
                46.00
              </div>
            </div>,
            "className": "productLabelContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="productLabel"
                >
                  Warehouse Side Split Roll Neck Jumper
                </div>,
                <div
                  className="productPrice"
                >
                  £
                  46.00
                </div>,
              ],
              "className": "centeringContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Warehouse Side Split Roll Neck Jumper",
                  "className": "productLabel",
                },
                "ref": null,
                "rendered": "Warehouse Side Split Roll Neck Jumper",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "£",
                    "46.00",
                  ],
                  "className": "productPrice",
                },
                "ref": null,
                "rendered": Array [
                  "£",
                  "46.00",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="imageCard"
          >
            <Link
              id="productImageId-123"
              onClick={[Function]}
              replace={false}
              to="products/123"
            >
              <img
                alt="Warehouse Side Split Roll Neck Jumper"
                src="./static/media/prod1-img1.jpg"
              />
            </Link>
          </div>,
          <Link
            className="divLink"
            id="productNameId-123"
            onClick={[Function]}
            replace={false}
            to="products/123"
          >
            <div
              className="productLabelContainer"
            >
              <div
                className="centeringContainer"
              >
                <div
                  className="productLabel"
                >
                  Warehouse Side Split Roll Neck Jumper
                </div>
                <div
                  className="productPrice"
                >
                  £
                  46.00
                </div>
              </div>
            </div>
          </Link>,
        ],
        "className": "ProductElement",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              id="productImageId-123"
              onClick={[Function]}
              replace={false}
              to="products/123"
            >
              <img
                alt="Warehouse Side Split Roll Neck Jumper"
                src="./static/media/prod1-img1.jpg"
              />
            </Link>,
            "className": "imageCard",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Warehouse Side Split Roll Neck Jumper"
                src="./static/media/prod1-img1.jpg"
              />,
              "id": "productImageId-123",
              "onClick": [Function],
              "replace": false,
              "to": "products/123",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Warehouse Side Split Roll Neck Jumper",
                "src": "./static/media/prod1-img1.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="productLabelContainer"
            >
              <div
                className="centeringContainer"
              >
                <div
                  className="productLabel"
                >
                  Warehouse Side Split Roll Neck Jumper
                </div>
                <div
                  className="productPrice"
                >
                  £
                  46.00
                </div>
              </div>
            </div>,
            "className": "divLink",
            "id": "productNameId-123",
            "onClick": [Function],
            "replace": false,
            "to": "products/123",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="centeringContainer"
              >
                <div
                  className="productLabel"
                >
                  Warehouse Side Split Roll Neck Jumper
                </div>
                <div
                  className="productPrice"
                >
                  £
                  46.00
                </div>
              </div>,
              "className": "productLabelContainer",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="productLabel"
                  >
                    Warehouse Side Split Roll Neck Jumper
                  </div>,
                  <div
                    className="productPrice"
                  >
                    £
                    46.00
                  </div>,
                ],
                "className": "centeringContainer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Warehouse Side Split Roll Neck Jumper",
                    "className": "productLabel",
                  },
                  "ref": null,
                  "rendered": "Warehouse Side Split Roll Neck Jumper",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "£",
                      "46.00",
                    ],
                    "className": "productPrice",
                  },
                  "ref": null,
                  "rendered": Array [
                    "£",
                    "46.00",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
